ArduCopter/GCS_Copter.cpp

```diff
@@ -106,6 +106,20 @@ void GCS_Copter::update_vehicle_sensor_status_flags(void)
         break;
     }
 
+    control_sensors_present |= MAV_SYS_STATUS_SENSOR_PROPULSION;
+    if (copter.arming.is_armed()) {
+        control_sensors_enabled |= MAV_SYS_STATUS_SENSOR_PROPULSION;
+    } else {
+        control_sensors_enabled &= ~MAV_SYS_STATUS_SENSOR_PROPULSION;
+    }
```

Comment:
Two thoughts:
 - perhaps a propulsion unit can be known-bad even when vehicle is not armed.
 - does this apply to servos with feedback which might be on fire?  In which case "propulsion" might not be great.  Did I previously suggest "actuator"?

Reply:
Future PRs can add ESC diagnostics while disarmed. But this one is not capable of doing that.
Servos are not tested, but it can be added in a future PR.

Reply:
> Future PRs can add ESC diagnostics while disarmed. But this one is not capable of doing that.

Disagree; the existing flag for motor lost doesn't get reset when the vehicle is disarmed.  So if you lose one in flight it should stay red on the ground, rather than moving to grey.

Reply:
Now I get your point. Yes, I will make it persistent.